### DISCUSSION TITLE
UserProfile Component

### DISCUSSION THREAD

**Discussion by aaron-hutton** [clerk/javascript#1144] [Support request - DO NOT OPEN HERE]
Hi, got some questions about the UserProfile component. I'm currently using NextJS app directory with `@clerk/nextjs` and `@clerk/nextjs/app-beta`.

1. Is there any way to just render the forms from the UserProfile, everything from `.cl-profilePage` down? ie [these components](https://github.com/clerkinc/javascript/tree/main/packages/clerk-js/src/ui/components/UserProfile). I'd like to embed the form into an existing settings page. Do I just have to use the `appearance` and `layout` props to try to remove all of the styling from these components (which is a pretty awful solution)?

2. If I can't use the individual components, does UserProfile path routing work in any way (client or server) with the app directory? [These docs](https://clerk.com/docs/users/user-profile) suggest it was possible with the pages directory but trying the same thing with the app directory is just giving me a blank card.

**Comment by jescalan** [comment]
https://github.com/orgs/clerk/discussions/3829
